Skeleton/src/Volunteer.cpp

```diff
@@ -1 +1,263 @@
-#include "../include/Volunteer.h";
+
+#include "../include/Volunteer.h"
+#include "../include/Order.h"
+
+
+/////////////////////////////////////////////////////////
+//////////////////////// Volunteer //////////////////////
+/////////////////////////////////////////////////////////
+
+Volunteer::Volunteer(int id, const string &name) : id(id), name(name) 
+{
+    completedOrderId = -1;
+    activeOrderId = -1;
+}
+
+Volunteer::Volunteer(int id, const string &name, int completedOrderId, int activeOrderId) : id(id), name(name), completedOrderId(completedOrderId), activeOrderId(activeOrderId) {}
+
+int Volunteer::getId() const
+{
+    return id;
+}
+
+const string& Volunteer::getName() const 
+{
+    return name;
+}
+
+int Volunteer::getActiveOrderId() const 
+{
+    return activeOrderId;
+}
+
+bool Volunteer::isBusy() const
+{
+    return (activeOrderId != NO_ORDER);
+}
+
+
+/////////////////////////////////////////////////////////////////////////////////
+//////////////////////////// Collector Volunteer ////////////////////////////////
+/////////////////////////////////////////////////////////////////////////////////
+
+
+CollectorVolunteer::CollectorVolunteer(int id, const string &name, const int coolDown) : Volunteer(id, name) ,coolDown(coolDown), timeLeft(NO_ORDER){}
+CollectorVolunteer* CollectorVolunteer::clone() const 
+{
+     return new CollectorVolunteer(*this);
+}
+void CollectorVolunteer::step() // need to check if it's correct
+{
+    if(decreaseCoolDown())
+    {
+        completedOrderId = activeOrderId;
+        activeOrderId = NO_ORDER;
+    }
+}
+int CollectorVolunteer::getCoolDown() const
+{
+    return coolDown;
+}
+
+int CollectorVolunteer::getTimeLeft() const 
+{
+    return timeLeft;
+}
+
+bool CollectorVolunteer::decreaseCoolDown() //Decrease timeLeft by 1,return true if timeLeft=0,false otherwise
```

Comment:
is it your comment or their comment,
if its their comment so no problem, if it is your comment- maybe if it was 0 before and call decrease again it should return false

Reply:
their comment